splunk/splunk_test.go

```diff
@@ -0,0 +1,64 @@
+package splunk
+
+import (
+	_ "embed"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/redhat-appstudio/segment-bridge.git/containerfixture"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed splunk_container_template.yaml
+var manifest string
+
+func TestGetRecordsCount(t *testing.T) {
+	_, callerFilePath, _, _ := runtime.Caller(0)
+	callerDirPath := filepath.Dir(callerFilePath)
+	netrcPath := fmt.Sprintf("%s/.netrc", callerDirPath)
+	fmt.Println(netrcPath)
+	
+
+	deployment := containerfixture.BuildAndRunPod(t, manifest)
+	endpoint := fmt.Sprintf("http://localhost:%s/services/server/status", deployment.ApiPort)
+	defer containerfixture.Cleanup(t, deployment.PodName)
+	containerfixture.VerifyServiceIsUp(t, endpoint, "splunk")
+	tests := []struct {
+		name       string
+		host       string
+		port       string
+		index      string
+		want       string
+		want_error bool
+	}{
+		{
+			name:  "Simple query",
+			host:  "localhost",
+			port:  deployment.ApiPort,
+			index: "test_index",
+			want:  "10",
+		},
+		{
+			name:       "Simple query with non-existing host",
+			host:       "fake-host",
```

Comment:
I don't see how is testing that you cannot query the wrong host useful for testing the fixture for bringing up the host... This is testing `GetRecordsCount`...

Reply:
removed for now, but I'm trying to address the errors in the code. since the host is a parameter, giving a wrong value for one of the parameters (anything but index or outputMode) will result in a failure. why not check that it works as expected?

Reply:
that would be a failure of the code talking to the fixture, not a failure of the fixture itself.